BUILDING.md

```diff
@@ -0,0 +1,99 @@
+## Building
+
+You need to set PICO_SDK_PATH in the environment, or pass it to cmake with `-DPICO_SDK_PATH=/path/to/pico-sdk`. To use features such as signing or hashing, you will need to make sure the mbedtls submodule in the SDK is checked out - this can be done by running this from your SDK directory.
+
+```console
+git submodule update --init lib/mbedtls
```

Comment:
Hmmm, in light of other recent changes, does this now also need to update mbedtls's submodules too?

Reply:
So now that **this** PR has been merged, does that imply that #226 ought to be rebased, and make its own changes to `README.md`? Or am I just getting confused by all the overlapping PRs again? :rofl:

Reply:
#226 removes the need for mbedtls submodules, so won't need the readme update

Reply:
Right, so does #226 also need to update `BUILDING.md` to remove the `git submodule update` command, or am I still getting my wires crossed?

Reply:
You still need the SDK's mbedtls submodule (pico-sdk/lib/mbedtls), but you don't need Mbedtls's submodules (pico-sdk/lib/mbedtls/framework etc) - so the command is still needed, but a new command to do it recursively instead is not needed, so no changes need to be made to the readme.

Reply:
Aha, thanks! It's the "you don't need Mbedtls's submodules (pico-sdk/lib/mbedtls/framework etc)" bit that I didn't realise, and is why I was getting confused :rofl:   But in that case, why is / was https://github.com/raspberrypi/pico-examples/pull/635 needed?

Reply:
Basically if you consider needing Mbedtls submodules to be the bug, then #226 is the bug-fix, and raspberrypi/pico-examples#635, https://github.com/raspberrypi/pico-sdk/pull/2416, 8be60e9eeab96857c84d71f0565cad0e84654c77 are all workarounds to get CI working